src/main/java/at/hannibal2/skyhanni/data/HotmData.kt

```diff
@@ -0,0 +1,628 @@
+            }
+
+            if (entry == PEAK_OF_THE_MOUNTAIN) {
+                entry.enabled = entry.activeLevel != 0
+                return
+            }
+            entry.enabled = lore.any { enabledPattern.matches(it) }
+        }
+
+        private fun Slot.handlePowder(): Boolean {
+            val item = this.stack ?: return false
+
+            val isHeartItem = when {
+                heartItemPattern.matches(item.name) -> true
+                resetItemPattern.matches(item.name) -> false
+                else -> return false
+            }
+
+            if (isHeartItem) { // Reset on the heart Item to remove duplication
+                tokens = 0
+                availableTokens = 0
+                HotmAPI.Powder.entries.forEach { it.reset() }
+                heartItem = this
+            }
+
+            val lore = item.getLore()
+
+            val tokenPattern = if (isHeartItem) heartTokensPattern else resetTokensPattern
+
+            lore@ for (line in lore) {
+
+                HotmAPI.Powder.entries.forEach {
+                    it.pattern(isHeartItem).matchMatcher(line) {
+                        val powder = group("powder").replace(",", "").toLong()
+                        if (isHeartItem) {
+                            it.setCurrent(powder)
+                        }
+                        it.addTotal(powder)
+                        continue@lore
+                    }
+                }
+
+                tokenPattern.matchMatcher(line) {
+                    val token = group("token").toInt()
+                    if (isHeartItem) {
+                        availableTokens = token
+                    }
+                    tokens += token
+                    continue@lore
+                }
+            }
+            return true
+        }
+
+        @SubscribeEvent
+        fun onScoreboardUpdate(event: ScoreboardChangeEvent) {
+            if (!LorenzUtils.inSkyBlock) return
+
+            event.newList.matchFirst(ScoreboardPattern.powderPattern) {
+                val type = HotmAPI.Powder.entries.firstOrNull { it.lowName == group("type") } ?: return
+                val amount = group("amount").formatLong()
+                val difference = amount - type.getCurrent()
+
+                if (difference > 0) {
+                    type.gain(difference)
+                    ChatUtils.debug("Gained §a${difference.addSeparators()} §e${type.lowName} Powder")
+                }
+            }
+        }
+
+        @SubscribeEvent
+        fun onInventoryClose(event: InventoryCloseEvent) {
+            if (!inInventory) return
+            inInventory = false
+            entries.forEach { it.slot = null }
+            heartItem = null
+        }
+
+        @SubscribeEvent
+        fun onInventoryFullyOpen(event: InventoryFullyOpenedEvent) {
+            if (!LorenzUtils.inSkyBlock) return
+            inInventory = inventoryPattern.matches(event.inventoryName)
+            DelayedRun.runNextTick {
```

Comment:
Why do we want to delay one tick here?

Reply:
Since the item ref is not the actual one. And it updates the inv on the next Tick

Reply:
ref?

Reply:
Do we really need the slot info? The only use case for the slot data I can find is `HotmData.heartItem`. And here we could also replace it to item stack. Then, we can use `event.inventoryItems` instead of `InventoryUtils.getItemsInOpenChest()` in InventoryFullyOpenedEvent.

Reply:
the slot is used in the rendering for the highlight. That is one use I can't remove easily. (It is HotmFeatures L;25)